app/derivative-services/vector_resource_derivative_service.rb

```diff
@@ -51,7 +51,11 @@ def cleanup_derivatives
     deleted_files = []
     vector_derivatives = resource.file_metadata.select { |file| file.derivative? || file.thumbnail_file? }
     vector_derivatives.each do |file|
-      storage_adapter.delete(id: file.id)
+      file_identifier = file.file_identifiers.try(:first)
```

Comment:
Maybe we should make `file_identifiers` not an optional array so we can just trust calling `.first`?